### PR TITLE
Update Segment identify calls to remove email, add domain

### PR DESCRIPTION
**Pull request by noeldevelops**
## Summary of Changes

- Instead of trying to send the full email to Segment, extract the domain ending and send that. 
- Created a new function specifically for sending a Segment Identify call that handles the details of parsing user data from the Session or UserInfo object. 
- Adds a call to this ^ function during the authentication part of Extension Activation to catch existing sessions. 
- ⚠️ Files changed looks bigger bc I refactored to split off `getTelemetryLogger` from the other functions to make testing easier (since mocking module-level functions is still a challenge) 

## Any additional details or context that should be provided?

- We found that VS Code TelemetryLogger class will [redact any email-looking string via RegEx](https://github.com/microsoft/vscode/blob/main/src/vs/platform/telemetry/common/telemetryUtils.ts#L326). We'd like to filter our final analytics data by internal (Confluent) users while respecting this PII, hence the addition of a `domain` property instead of `email`
- Here's the event as it appears in Segment: 
Screenshot 2024-10-22 at 2 54 32 PM

## Pull request checklist

Please check if your PR fulfills the following (if applicable):

##### Tests

- [X] Added new
- [ ] Updated existing
- [ ] Deleted existing

##### Other

- [ ] Does anything in this PR need to be mentioned in the user-facing [CHANGELOG](https://github.com/confluentinc/vscode/blob/main/CHANGELOG.md) or [README](https://github.com/confluentinc/vscode/blob/main/public/README.md)?
- [ ] Have you validated this change locally by [packaging](https://github.com/confluentinc/vscode/blob/main/README.md#packaging-steps) and installing the extension `.vsix` file?
  ```shell
  gulp clicktest
  ```

### CODE REVIEW

File: src/telemetry/telemetry.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  userInfo: UserInfo | undefined;
+  session: vscode.AuthenticationSession | undefined;
+}) {
+  const id = userInfo?.id || session?.account.id;
+  const username = userInfo?.username || session?.account.label;
+  const social_connection = userInfo?.social_connection;
+  let domain: string | undefined;
+  if (username) {
+    //  email is redacted by VSCode TelemetryLogger, but we extract domain for Confluent analytics use
+    const emailRegex = /@[a-zA-Z0-9-]+\.[a-zA-Z0-9-]+/;
```
Comment (MSeal) [inline src/telemetry/telemetry.ts:29]: I'd of maybe just done if `@` present, then split and grab `[1]` since we don't care about other properties and expect it to always be an email entry anyway.

File: src/telemetry/telemetryLogger.ts [hunk 2]
```diff
@@ -65,25 +59,28 @@ export function getTelemetryLogger(): vscode.TelemetryLogger {
...
+          anonymousId: segmentAnonId,
+          traits: { ...data.user },
         });
         // We don't want to send the user traits or identify prop in the following Track call
         delete data.identify;
         delete data.user;
       }
       analytics?.track({
         userId,
-        anonymousId: data?.["common.vscodesessionid"] || segmentAnonId,
```
Comment (MSeal) [inline src/telemetry/telemetryLogger.ts:86]: Why this change?

**Review by MSeal — APPROVED** [review]
Some minor questions, good for merge independently imo.

File: src/telemetry/telemetryLogger.ts (hunk 2 above)
Comment (noeldevelops) [reply src/telemetry/telemetryLogger.ts:86]: Just setting us up to handle logout in near future... We're already sending `vscodesessionid` with the common properties in data, and the `segmentAnonId` is stable in each instance of the TelemetryLogger, so it won't make much difference now but in updates I'll be able to reset the anon id (during log out flow or other cases where the User Id might be new for an existing VSCode session)

File: src/telemetry/telemetry.ts (hunk 1 above)
Comment (noeldevelops) [reply src/telemetry/telemetry.ts:29]: does sound simpler! I wasn't sure if we were 100% certain this will be email every time. Happy to update if it makes sense to do so in follow ups
